feature/interface/aggregate/otg_tests/aggregate_all_not_viable_test/aggregate_all_not_forwarding_viable_test.go

```diff
@@ -0,0 +1,1025 @@
+	for _, port := range dutPortList {
+		rxPktsBeforeTraffic[port] = gnmi.Get(t, dut, gnmi.OC().Interface(port.Name()).Counters().InPkts().State())
+		txPktsBeforeTraffic[port] = gnmi.Get(t, dut, gnmi.OC().Interface(port.Name()).Counters().OutPkts().State())
+	}
+}
+
+// verifyTrafficFlow verify the each flow on ATE
+func verifyTrafficFlow(t *testing.T, ate *ondatra.ATEDevice, flows []gosnappi.Flow) bool {
+	if flows[0].Name() == "pfx1ToPfx4" {
+		rxPkts := gnmi.Get(t, ate.OTG(), gnmi.OTG().Flow(flows[0].Name()).Counters().InPkts().State())
+		if got := rxPkts / 100; got == 0 {
+			t.Logf("Packet Dropped, LossPct for flow %s: got %d, want 0 packet", flows[0].Name(), got)
+		} else {
+			return false
+		}
+	} else {
+		for _, flow := range flows {
+			rxPkts := gnmi.Get(t, ate.OTG(), gnmi.OTG().Flow(flow.Name()).Counters().InPkts().State())
+			txPkts := gnmi.Get(t, ate.OTG(), gnmi.OTG().Flow(flow.Name()).Counters().OutPkts().State())
+			lostPkt := txPkts - rxPkts
+			if got := (lostPkt * 100 / txPkts); got > 0 {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+// awaitAdjacency wait for adjacency to be up/down
+func awaitAdjacency(t *testing.T, dut *ondatra.DUTDevice, intfName string, state oc.E_Isis_IsisInterfaceAdjState) bool {
+	isisPath := ocpath.Root().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_ISIS, isisInstance).Isis()
+	intf := isisPath.Interface(intfName)
+	query := intf.LevelAny().AdjacencyAny().AdjacencyState().State()
+	_, ok := gnmi.WatchAll(t, dut, query, time.Minute, func(val *ygnmi.Value[oc.E_Isis_IsisInterfaceAdjState]) bool {
+		v, ok := val.Val()
+		return v == state && ok
+	}).Await(t)
+
+	return ok
+
+}
+
+// checkBidirectionalTraffic verify the bidirectional traffic on DUT ports.
+func checkBidirectionalTraffic(t *testing.T, dut *ondatra.DUTDevice, portList []*ondatra.Port) error {
+
+	for _, port := range portList {
+		txPkts := gnmi.Get(t, dut, gnmi.OC().Interface(port.Name()).Counters().OutPkts().State())
+		rxPkts := gnmi.Get(t, dut, gnmi.OC().Interface(port.Name()).Counters().InPkts().State())
+		if got := (rxPkts - rxPktsBeforeTraffic[port]) / 100; got == 0 {
+			return fmt.Errorf("No Packet received, LossPct on Port %s: got %d", port.Name(), got)
+		}
+		if got := (txPkts - txPktsBeforeTraffic[port]) / 100; got == 0 {
+			return fmt.Errorf("No Packet transmitted, LossPct on Port %s: got %d", port.Name(), got)
+		}
+	}
+	return nil
+}
+
+// confirmNonViableForwardingTraffic verify the traffic received on DUT
+// interfaces and transmitted to ATE-1
+func confirmNonViableForwardingTraffic(t *testing.T, dut *ondatra.DUTDevice, ate *ondatra.ATEDevice,
+	atePort []*ondatra.Port, dutPort []*ondatra.Port) error {
+
+	// Ensure no traffic is transmitted out of DUT ports with Forwarding Viable False
+	for _, port := range atePort {
+		rxPkts := gnmi.Get(t, ate.OTG(), gnmi.OTG().Port(port.ID()).Counters().InFrames().State())
+		if got := rxPkts / 100; got > 0 {
```

Comment:
same here , there is no need to % calculation

Reply:
Here i am not calculating any % calculation, Just verifying if there is no packet rx/tx on the DUT port throw an error.
Note: i am using (txPkts - txPktsBeforeTraffic[port]) /100 because there will be some packet for rx/tx (ARP, ISIS, BGP) which is always <100.

Reply:
Perhaps more accurate and deterministic to check the ATE flow stats instead of the ATE port stats?   Does OTG support getting flow stats per ATE port or similar?